// hebrew-transliteration/dist/schema.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SBL = exports.Schema = void 0;
/**
 * class for defining a schema for transliteration
 */
class Schema {
    constructor(schema) {
        (this.VOCAL_SHEVA = schema.VOCAL_SHEVA),
            (this.HATAF_SEGOL = schema.HATAF_SEGOL),
            (this.HATAF_PATAH = schema.HATAF_PATAH),
            (this.HATAF_QAMATS = schema.HATAF_QAMATS),
            (this.HIRIQ = schema.HIRIQ),
            (this.TSERE = schema.TSERE),
            (this.SEGOL = schema.SEGOL),
            (this.PATAH = schema.PATAH),
            (this.QAMATS = schema.QAMATS),
            (this.HOLAM = schema.HOLAM),
            (this.QUBUTS = schema.QUBUTS),
            (this.DAGESH = schema.DAGESH),
            (this.DAGESH_CHAZAQ = schema.DAGESH_CHAZAQ),
            (this.MAQAF = schema.MAQAF),
            (this.PASEQ = schema.PASEQ),
            (this.SOF_PASUQ = schema.SOF_PASUQ),
            (this.QAMATS_QATAN = schema.QAMATS_QATAN),
            (this.FURTIVE_PATAH = schema.FURTIVE_PATAH),
            (this.HIRIQ_YOD = schema.HIRIQ_YOD),
            (this.TSERE_YOD = schema.TSERE_YOD),
            (this.SEGOL_YOD = schema.SEGOL_YOD),
            (this.SHUREQ = schema.SHUREQ),
            (this.HOLAM_VAV = schema.HOLAM_VAV),
            (this.QAMATS_HE = schema.QAMATS_HE),
            (this.SEGOL_HE = schema.SEGOL_HE),
            (this.TSERE_HE = schema.TSERE_HE),
            (this.MS_SUFX = schema.MS_SUFX),
            (this.ALEF = schema.ALEF),
            (this.BET_DAGESH = schema.BET_DAGESH),
            (this.BET = schema.BET),
            (this.GIMEL = schema.GIMEL),
            (this.GIMEL_DAGESH = schema.GIMEL_DAGESH),
            (this.DALET = schema.DALET),
            (this.DALET_DAGESH = schema.DALET_DAGESH),
            (this.HE = schema.HE),
            (this.VAV = schema.VAV),
            (this.ZAYIN = schema.ZAYIN),
            (this.HET = schema.HET),
            (this.TET = schema.TET),
            (this.YOD = schema.YOD),
            (this.FINAL_KAF = schema.FINAL_KAF),
            (this.KAF = schema.KAF),
            (this.KAF_DAGESH = schema.KAF_DAGESH),
            (this.LAMED = schema.LAMED),
            (this.FINAL_MEM = schema.FINAL_MEM),
            (this.MEM = schema.MEM),
            (this.FINAL_NUN = schema.FINAL_NUN),
            (this.NUN = schema.NUN),
            (this.SAMEKH = schema.SAMEKH),
            (this.AYIN = schema.AYIN),
            (this.FINAL_PE = schema.FINAL_PE),
            (this.PE = schema.PE),
            (this.PE_DAGESH = schema.PE_DAGESH),
            (this.FINAL_TSADI = schema.FINAL_TSADI),
            (this.TSADI = schema.TSADI),
            (this.QOF = schema.QOF),
            (this.RESH = schema.RESH),
            (this.SHIN = schema.SHIN),
            (this.SIN = schema.SIN),
            (this.TAV = schema.TAV),
            (this.TAV_DAGESH = schema.TAV_DAGESH),
            (this.DIVINE_NAME = schema.DIVINE_NAME),
            (this.SYLLABLE_SEPARATOR = schema.SYLLABLE_SEPARATOR),
            (this.ADDITIONAL_FEATURES = schema.ADDITIONAL_FEATURES),
            (this.STRESS_MARKER = schema.STRESS_MARKER),
            (this.longVowels = schema.longVowels),
            (this.qametsQatan = schema.qametsQatan),
            (this.sqnmlvy = schema.sqnmlvy),
            (this.wawShureq = schema.wawShureq),
            (this.article = schema.article);
        this.allowNoNiqqud = schema.allowNoNiqqud;
    }
}
exports.Schema = Schema;
class SBL extends Schema {
    constructor(schema) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p, _q, _r, _s, _t, _u, _v, _w, _x, _y, _z, _0, _1, _2, _3, _4, _5, _6, _7, _8, _9, _10, _11, _12, _13, _14, _15, _16, _17, _18, _19, _20, _21, _22, _23, _24, _25, _26, _27, _28, _29, _30, _31, _32, _33, _34, _35, _36, _37, _38, _39, _40, _41, _42, _43, _44, _45, _46;
        super({
            VOCAL_SHEVA: (_a = schema.VOCAL_SHEVA) !== null && _a !== void 0 ? _a : "ǝ",
            HATAF_SEGOL: (_b = schema.HATAF_SEGOL) !== null && _b !== void 0 ? _b : "ĕ",
            HATAF_PATAH: (_c = schema.HATAF_PATAH) !== null && _c !== void 0 ? _c : "ă",
            HATAF_QAMATS: (_d = schema.HATAF_QAMATS) !== null && _d !== void 0 ? _d : "ŏ",
            HIRIQ: (_e = schema.HIRIQ) !== null && _e !== void 0 ? _e : "i",
            TSERE: (_f = schema.TSERE) !== null && _f !== void 0 ? _f : "ē",
            SEGOL: (_g = schema.SEGOL) !== null && _g !== void 0 ? _g : "e",
            PATAH: (_h = schema.PATAH) !== null && _h !== void 0 ? _h : "a",
            QAMATS: (_j = schema.QAMATS) !== null && _j !== void 0 ? _j : "ā",
            HOLAM: (_k = schema.HOLAM) !== null && _k !== void 0 ? _k : "ō",
            QUBUTS: (_l = schema.QUBUTS) !== null && _l !== void 0 ? _l : "ū",
            DAGESH: (_m = schema.DAGESH) !== null && _m !== void 0 ? _m : "",
            DAGESH_CHAZAQ: (_o = schema.DAGESH_CHAZAQ) !== null && _o !== void 0 ? _o : true,
            MAQAF: (_p = schema.MAQAF) !== null && _p !== void 0 ? _p : "-",
            PASEQ: (_q = schema.PASEQ) !== null && _q !== void 0 ? _q : "",
            SOF_PASUQ: (_r = schema.SOF_PASUQ) !== null && _r !== void 0 ? _r : "",
            QAMATS_QATAN: (_s = schema.QAMATS_QATAN) !== null && _s !== void 0 ? _s : "o",
            FURTIVE_PATAH: (_t = schema.FURTIVE_PATAH) !== null && _t !== void 0 ? _t : "a",
            HIRIQ_YOD: (_u = schema.HIRIQ_YOD) !== null && _u !== void 0 ? _u : "î",
            TSERE_YOD: (_v = schema.TSERE_YOD) !== null && _v !== void 0 ? _v : "ê",
            SEGOL_YOD: (_w = schema.SEGOL_YOD) !== null && _w !== void 0 ? _w : "ê",
            SHUREQ: (_x = schema.SHUREQ) !== null && _x !== void 0 ? _x : "û",
            HOLAM_VAV: (_y = schema.HOLAM_VAV) !== null && _y !== void 0 ? _y : "ô",
            QAMATS_HE: (_z = schema.QAMATS_HE) !== null && _z !== void 0 ? _z : "â",
            SEGOL_HE: (_0 = schema.SEGOL_HE) !== null && _0 !== void 0 ? _0 : "ê",
            TSERE_HE: (_1 = schema.TSERE_HE) !== null && _1 !== void 0 ? _1 : "ê",
            MS_SUFX: (_2 = schema.MS_SUFX) !== null && _2 !== void 0 ? _2 : "āyw",
            ALEF: (_3 = schema.ALEF) !== null && _3 !== void 0 ? _3 : "ʾ",
            BET: (_4 = schema.BET) !== null && _4 !== void 0 ? _4 : "b",
            BET_DAGESH: (_5 = schema.BET_DAGESH) !== null && _5 !== void 0 ? _5 : undefined,
            GIMEL: (_6 = schema.GIMEL) !== null && _6 !== void 0 ? _6 : "g",
            GIMEL_DAGESH: (_7 = schema.GIMEL_DAGESH) !== null && _7 !== void 0 ? _7 : undefined,
            DALET: (_8 = schema.DALET) !== null && _8 !== void 0 ? _8 : "d",
            DALET_DAGESH: (_9 = schema.DALET_DAGESH) !== null && _9 !== void 0 ? _9 : undefined,
            HE: (_10 = schema.HE) !== null && _10 !== void 0 ? _10 : "h",
            VAV: (_11 = schema.VAV) !== null && _11 !== void 0 ? _11 : "w",
            ZAYIN: (_12 = schema.ZAYIN) !== null && _12 !== void 0 ? _12 : "z",
            HET: (_13 = schema.HET) !== null && _13 !== void 0 ? _13 : "ḥ",
            TET: (_14 = schema.TET) !== null && _14 !== void 0 ? _14 : "ṭ",
            YOD: (_15 = schema.YOD) !== null && _15 !== void 0 ? _15 : "y",
            FINAL_KAF: (_16 = schema.FINAL_KAF) !== null && _16 !== void 0 ? _16 : "k",
            KAF: (_17 = schema.KAF) !== null && _17 !== void 0 ? _17 : "k",
            KAF_DAGESH: (_18 = schema.KAF_DAGESH) !== null && _18 !== void 0 ? _18 : undefined,
            LAMED: (_19 = schema.LAMED) !== null && _19 !== void 0 ? _19 : "l",
            FINAL_MEM: (_20 = schema.FINAL_MEM) !== null && _20 !== void 0 ? _20 : "m",
            MEM: (_21 = schema.MEM) !== null && _21 !== void 0 ? _21 : "m",
            FINAL_NUN: (_22 = schema.FINAL_NUN) !== null && _22 !== void 0 ? _22 : "n",
            NUN: (_23 = schema.NUN) !== null && _23 !== void 0 ? _23 : "n",
            SAMEKH: (_24 = schema.SAMEKH) !== null && _24 !== void 0 ? _24 : "s",
            AYIN: (_25 = schema.AYIN) !== null && _25 !== void 0 ? _25 : "ʿ",
            FINAL_PE: (_26 = schema.FINAL_PE) !== null && _26 !== void 0 ? _26 : "p",
            PE: (_27 = schema.PE) !== null && _27 !== void 0 ? _27 : "p",
            PE_DAGESH: (_28 = schema.PE_DAGESH) !== null && _28 !== void 0 ? _28 : undefined,
            FINAL_TSADI: (_29 = schema.FINAL_TSADI) !== null && _29 !== void 0 ? _29 : "ṣ",
            TSADI: (_30 = schema.TSADI) !== null && _30 !== void 0 ? _30 : "ṣ",
            QOF: (_31 = schema.QOF) !== null && _31 !== void 0 ? _31 : "q",
            RESH: (_32 = schema.RESH) !== null && _32 !== void 0 ? _32 : "r",
            SHIN: (_33 = schema.SHIN) !== null && _33 !== void 0 ? _33 : "š",
            SIN: (_34 = schema.SIN) !== null && _34 !== void 0 ? _34 : "ś",
            TAV: (_35 = schema.TAV) !== null && _35 !== void 0 ? _35 : "t",
            TAV_DAGESH: (_36 = schema.TAV_DAGESH) !== null && _36 !== void 0 ? _36 : undefined,
            DIVINE_NAME: (_37 = schema.DIVINE_NAME) !== null && _37 !== void 0 ? _37 : "yhwh",
            SYLLABLE_SEPARATOR: (_38 = schema.SYLLABLE_SEPARATOR) !== null && _38 !== void 0 ? _38 : undefined,
            ADDITIONAL_FEATURES: (_39 = schema.ADDITIONAL_FEATURES) !== null && _39 !== void 0 ? _39 : undefined,
            STRESS_MARKER: (_40 = schema.STRESS_MARKER) !== null && _40 !== void 0 ? _40 : undefined,
            longVowels: (_41 = schema.longVowels) !== null && _41 !== void 0 ? _41 : true,
            qametsQatan: (_42 = schema.qametsQatan) !== null && _42 !== void 0 ? _42 : true,
            sqnmlvy: (_43 = schema.sqnmlvy) !== null && _43 !== void 0 ? _43 : true,
            wawShureq: (_44 = schema.wawShureq) !== null && _44 !== void 0 ? _44 : true,
            article: (_45 = schema.article) !== null && _45 !== void 0 ? _45 : true,
            allowNoNiqqud: (_46 = schema.allowNoNiqqud) !== null && _46 !== void 0 ? _46 : true
        });
    }
}
exports.SBL = SBL;
